src/environment.rs

```diff
@@ -57,25 +57,15 @@ fn create_prefix_location_file(prefix_file: &Path) -> miette::Result<()> {
 ///     1. It verifies that the prefix location is unchanged.
 ///     2. It verifies that the project supports the current platform.
 ///     3. It verifies that the system requirements are met.
-pub fn sanity_check_project(project: &Project) -> miette::Result<()> {
+pub fn sanity_check_project(project: &Project, no_install: &mut bool) -> miette::Result<()> {
     // Sanity check of prefix location
     verify_prefix_location_unchanged(project.pixi_dir().join(consts::PREFIX_FILE_NAME).as_path())?;
 
     // Make sure the project supports the current platform
     let platform = Platform::current();
     if !project.platforms().contains(&platform) {
-        let span = project.manifest.parsed.project.platforms.span();
-        return Err(miette::miette!(
-            help = format!(
-                "The project needs to be configured to support your platform ({platform})."
-            ),
-            labels = vec![LabeledSpan::at(
-                span.unwrap_or_default(),
-                format!("add '{platform}' here"),
-            )],
-            "the project is not configured for your current platform"
-        )
-        .with_source_code(project.manifest_named_source()));
+        *no_install = true;
+        tracing::warn!("Adding dependency for unsupported platform ({platform}).")
```

Comment:
It think it would be better to redo this as 
```suggestion
        tracing::warn!("Not installing dependency on current platform: ({platform}) as it is not part of the supported platforms. 
```